src/main/webapp/app/exam/participate/exam-navigation-sidebar/exam-navigation-sidebar.component.ts

```diff
@@ -0,0 +1,254 @@
+            }
+        });
+
+        const isInitialSession = this.examSessions && this.examSessions.length > 0 && this.examSessions[0].initialSession;
+        if (isInitialSession || isInitialSession == undefined) {
+            return;
+        }
+
+        // If it is not an initial session, update the isSynced variable for out of sync submissions.
+        this.exercises
+            .filter((exercise) => exercise.type === ExerciseType.PROGRAMMING && exercise.studentParticipations)
+            .forEach((exercise) => {
+                const domain: DomainChange = [DomainType.PARTICIPATION, exercise.studentParticipations![0]];
+                this.conflictService.setDomain(domain);
+                this.repositoryService.setDomain(domain);
+
+                this.repositoryService
+                    .getStatus()
+                    .pipe(map((response) => Object.values(CommitState).find((commitState) => commitState === response.repositoryStatus)))
+                    .subscribe((commitState) => {
+                        const submission = ExamParticipationService.getSubmissionForExercise(exercise);
+                        if (commitState === CommitState.UNCOMMITTED_CHANGES && submission) {
+                            // If there are uncommitted changes: set isSynced to false.
+                            submission.isSynced = false;
+                        }
+                    });
+            });
+
+        this.refreshExerciseSaveCount();
+    }
+
+    ngOnDestroy() {
+        this.profileSubscription?.unsubscribe();
+        this.sidebarEventService.emitResetValue();
+    }
+
+    getExerciseButtonTooltip(exercise: Exercise): ButtonTooltipType {
+        return this.examParticipationService.getExerciseButtonTooltip(exercise);
+    }
+
+    getExerciseIconTooltip(exercise: Exercise): string {
+        return getIconTooltip(exercise.type);
+    }
+
+    /**
+     * @param overviewPage: user wants to switch to the overview page
+     * @param exerciseIndex: index of the exercise to switch to, if it should not be used, you can pass -1
+     * @param forceSave: true if forceSave shall be used.
+     * @param submission the submission to be viewed, used in the exam timeline
+     */
+    changePage(overviewPage: boolean, exerciseIndex: number, forceSave?: boolean, submission?: SubmissionVersion | ProgrammingSubmission | FileUploadSubmission): void {
+        if (!overviewPage) {
+            // out of index -> do nothing
+            if (exerciseIndex > this.exercises.length - 1 || exerciseIndex < 0) {
+                return;
+            }
+            // set index and emit event
+            this.exerciseIndex = exerciseIndex;
+            this.onPageChanged.emit({ overViewChange: false, exercise: this.exercises[this.exerciseIndex], forceSave: !!forceSave, submission: submission });
+        } else if (overviewPage) {
+            // set index and emit event
+            this.exerciseIndex = -1;
```

Comment:
Could we make -1  a variable, e.g. "NOT_INCLUDED_INDEX" and link that in the javadoc instead?